include/parpeoptimization/minibatchOptimization.h

```diff
@@ -645,17 +705,35 @@ class MinibatchOptimizer {
             if (cost2 > cost1) {
                 /* The parabola idea didn't work. Just admit the step, as it is */
                 cost1 = evalLineSearch(stepLength);
+                parabola_ss << " Step adaption did not work..." << std::endl;
             }
+            parabola_ss << "   cost0: " << cost << " (steplength: " << getVectorNorm(parDifference)
+                << "),   cost1: " << cost1 << " (steplength: " << stepLength
+                << "),   cost2: " << cost2 << " (steplength: " << newStepLength
+                << ") " << std::endl;
+            if (logger)
+                logger->logmessage(LOGLVL_INFO, parabola_ss.str().c_str());
+            
             /* We tried all we could */
             return;
         }
         
         /* Original step was too big, but we're facing a descent direction 
          * Propose a new step based on a parabolic interpolation */
-        double newStepLength = 0.5 * dirGradient * std::pow(stepLength, 2.0) /
+        double newStepLength = -0.5 * dirGradient * std::pow(stepLength, 2.0) /
                 (cost1 - cost - dirGradient * stepLength);
         double cost2 = evalLineSearch(newStepLength);
         
+        // Debugging output
+        if (logger) {
+            line_ss << " Line-Search: two steps were done, results are: " << std::endl
+                << "   cost0: " << cost << " (steplength: " << getVectorNorm(parDifference)
```

Comment:
What are cost0, cost1, cost2? Pls document

Reply:
I wrote a longer comment on what these quantities are and what they do in lines 668 - 680. Is it okay like this or do you want a more formal documentation?

Reply:
Works for me.